src/E225B0.c

```diff
@@ -1,7 +1,203 @@
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+            // spA4 = &D_802B7BA0_E23150;
+            command.unk_04 = &D_802B7BA0_E23150;
+            break;
+        case 4:
+        case 5:
+        case 6:
+        case 7:
+            // spA4 = &D_802B7BC0_E23170;
+            command.unk_04 = &D_802B7BC0_E23170;
+            break;
+        case 8:
+        case 9:
+        case 10:
+        case 11:
+            // spA4 = &D_802B7BE0_E23190;
+            command.unk_04 = &D_802B7BE0_E23190;
+            break;
+        }
+        func_8013AA9C(0, 7, 0xFF, 0xFF, 0xFF, D_802B7C78_E23228->unk_28, 0);
+
+        command.unk_00 = &D_802B7580_E22B30;
+        command.unk_08 = 0x38;
+        command.unk_0A = 0x38;
+        command.unk_0C = -0x1C;
+        command.unk_0E = 0x2E;
+        command.unk_10 = 0xFF;
+
+        func_8013AF70(0, &command, 0, &matrix2);
+        gSPPopMatrix(gMasterGfxPos++, 0);
+    }
+}
+
+
+void func_802B72C0_E22870(void) {
+    mem_clear((s8*)D_802B7C78_E23228, 0x2C);
+
+    D_802B7C78_E23228->x = (f32)gPlayerStatus.position.x;
+    D_802B7C78_E23228->y = (f32)(gPlayerStatus.position.y + (f32)gPlayerStatus.colliderHeight + 8.0f);
+    D_802B7C78_E23228->z = (f32)gPlayerStatus.position.z;
+
+    D_802B7C78_E23228->unk_28 = 0xFF;
+
+    gPlayerStatus.animFlags |= 0x100;
+    D_8010C93C = &func_802B735C_E2290C;
+}
+
+
+typedef struct struct8015A578 {
+    u8 unk_0[0x2];
+    u8 unk_2;
+    u8 unk_3[0x9];
+} struct8015A578;
+extern struct8015A578 D_8015A578;
+
+void func_802B735C_E2290C(void) {
+    f32 temp_f2;
+    s32 temp_v0_3;
+    s32 temp_v1;
+    struct802B7C78* temp_a0;
+    struct802B7C78* temp_v1_3;
+    s32 phi_v0;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+
+    temp_a0 = D_802B7C78_E23228;
+    temp_f2 = temp_a0->y;
+    temp_v1 = temp_a0->unk_24;
+    temp_a0->y = temp_f2 + (((playerStatus->position.y + (f32) playerStatus->colliderHeight + 10.0f)
+            - temp_f2) / 1.5f);
+    temp_a0->x = playerStatus->position.x;
+    temp_a0->z = playerStatus->position.z;
+
+    switch (temp_v1) {
+    case 0:
+        if (partnerActionStatus->actionState.b[0] != 0
+                && partnerActionStatus->actionState.b[3] == 8) {
+            phi_v0 = gGameStatusPtr->unk_7D;
+        } else {
+            phi_v0 = playerStatus->flags & 0x3000;
+        }
+        if (phi_v0 == 0) {
+            temp_v1_3 = D_802B7C78_E23228;
+            temp_v1_3->unk_24 += 1;
+            return;
+        }
+        break;
+
+    case 1:
+        if ((playerStatus->flags & 0x20) != 0) {
+            temp_a0->unk_24 = 3;
+            return;
+        }
+
+        if (temp_a0->unk_18++ >= 0x10) {
+            temp_a0->scale = 0.36f;
+            temp_a0->unk_24 += 1;
+        }
+        break;
+
+    case 2:
+        temp_a0->scale = 0.57f;
+        temp_a0->unk_24 += 1;
+        sfx_play_sound_at_player(0x17B, 0);
```

Comment:
in enums.h there's a sound effect enum - if this one exists, please use the value there, and if not, please create one for 0x17B